lib/python/pyflyby/_importstmt.py

```diff
@@ -486,11 +487,31 @@ def pretty_print(self, params=FormatParams(),
             tokens.append(t)
         res = s0 + pyfill(s, tokens, params=params)
         if params.use_black:
-            import black
-            mode = black.FileMode()
-            return black.format_str(res, mode=mode)
+            return self.run_black(res, params)
         return res
 
+    @staticmethod
+    def run_black(str_to_format, params):
+        black_cmd = [
+            f'black --line-length {str(params.max_line_length)} -c "{str_to_format.strip()}"',
+        ]
+        try:
+            completed_process = subprocess.run(
+                black_cmd,
+                shell=True,
+                stdout=subprocess.PIPE,
+                stderr=subprocess.PIPE,
+                universal_newlines=True,
+            )
```

Comment:
You likely should not do that, in general every-time you see `shell=True` you should step back and think security. 

The first argument to `run` is a list of your program arguments, unless you **really** want them parsed by the shell. And Here you are forming a string to get it parsed, while you could have `['black', '--line-length', 'str(params.max_line_length)', ...]`.

This is :

1) Safer security wise (no risk to have shell escape, or alike
2) More correct (no risk of bas escape in `str_to_format`).